lib/src/models/parsed_token.dart

```diff
@@ -1,13 +1,18 @@
 import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:ledger_cardano/src/utils/constants.dart';
+import 'package:ledger_cardano/src/utils/utilities.dart';
 
 part 'parsed_token.freezed.dart';
 
 @freezed
-sealed class ParsedToken<T> with _$ParsedToken<T> {
-  ParsedToken._();
+sealed class ParsedToken with _$ParsedToken {
+  ParsedToken._() {
+    validateInt64(amount, 'amount');
+    validateMaxHexString(assetNameHex, 'assetNameHex', maxHexStringLength);
```

Comment:
could also validate asset name is `hex` characters only

Reply:
`maxHexStringLength` should better be named sth like `stringLength64Bits`. The current name implies that hex strings cannot be longer than that in general, which is not true.

`validateMaxHexString` should be renamed to `validateMaxStringLength` (and no mention of `hex` unless you actually verify if it's only hex characters too)

Reply:
done